dot/peerset/peerset.go

```diff
@@ -690,10 +690,12 @@ func (ps *PeerSet) disconnect(setIdx int, reason DropReason, peers ...peer.ID) e
 			return fmt.Errorf("cannot disconnect: %w", err)
 		}
 
-		ps.resultMsgCh <- Message{
-			Status: Drop,
-			setID:  uint64(setIdx),
-			PeerID: pid,
+		if ps.resultMsgCh != nil {
+			ps.resultMsgCh <- Message{
+				Status: Drop,
+				setID:  uint64(setIdx),
+				PeerID: pid,
+			}
```

Comment:
Shouldn't the producer close the channel? So somewhere here :thinking:

Reply:
I think there are problems with the arch of `peerset.Handler` that we should address at some point.  
- Right now this channel is returned by `Handler.Messages()`, but it's not a true pipeline.  If `handler.Messages()` was called again and was consumed, then messages would be split among all readers.  
- IMO the producer is `peerset.Peerset` so the producer is actually closing the channel.  Just on another goroutine.